_overviews/overview/scala-tools.md

```diff
@@ -0,0 +1,320 @@
+- [Maven](https://maven.apache.org/)
+- [Mill](https://www.lihaoyi.com/mill/)
+
+
+
+## Using sbt with ScalaTest
+
+ScalaTest is one of the main testing libraries for Scala projects, and in this section you’ll see how to create a Scala/sbt project that uses ScalaTest.
+
+
+### Creating the project directory structure
+
+As with the previous lesson, create an sbt project directory structure for a project named *HelloScalaTest* with the following commands:
+
+```sh
+mkdir HelloScalaTest
+cd HelloScalaTest
+mkdir -p src/{main,test}/{resources,scala}
+mkdir project target
+```
+
+
+### Creating the build.sbt file
+
+Next, create a *build.sbt* file in the root directory of your project with these contents:
+
+```scala
+name := "HelloScalaTest"
+version := "1.0"
+scalaVersion := "{{site.scala-version}}"
+
+libraryDependencies ++= Seq(
+  "org.scalatest" % "scalatest-core_0.27" % "3.2.2",
+  "org.scalatest" % "scalatest_0.27" % "3.2.2" % Test
+)
+```
+
+The first three lines of this file are essentially the same as the first example. The `libraryDependencies` lines tell sbt to include the dependencies (JAR files) that are needed to include ScalaTest.
+
+>The ScalaTest documentation has always been good, and you can always find the up to date information on what those lines should look like on the [Installing ScalaTest](https://www.scalatest.org/install) page.
+
+
+### Create a Scala source code file
+
+Next, create a Scala program that you can use to demonstrate ScalaTest. First, create a directory under *src/main/scala* named *math*:
+
+```sh
+$ mkdir src/main/scala/math
+            ----
+```
+
+Then, inside that directory, create a file named *MathUtils.scala* with these contents:
+
+```scala
+package math
+
+object MathUtils:
+  def double(i: Int) = i * 2
+```
+
+There isn’t much that can go wrong with that source code, but it provides a simple way to demonstrate ScalaTest. This object doesn’t have a `main` method, so instead of trying to run the project with `sbt run`, we’ll just compile it with `sbt compile`:
+
+````
+$ sbt compile
+
+[info] welcome to sbt
+[info] loading settings for project ...
+[info] loading project definition ...
+[info] loading settings for project ...
+[info] Executing in batch mode. For better performance use sbt's shell
+[success] Total time: 1 s
+````
+
+With that compiling, let’s create a ScalaTest file to test the `double` method.
+
+
+### Your first ScalaTest tests
+
+ScalaTest is very flexible, and there are several different ways to write tests. A simple way to get started is to write tests using the ScalaTest `AnyFunSuite`. To get started, create a directory named *math* under the *src/test/scala* directory, like this:
+
+```sh
+$ mkdir src/test/scala/math
+            ----
+```
+
+Next, create a file named *MathUtilsTests.scala* in that directory with the following contents:
+
+```scala
+package math
+  
+import org.scalatest.funsuite.AnyFunSuite
+
+class MathUtilsTests extends AnyFunSuite {
```

Comment:
do you want to update this to indentation syntax?

Reply:
I just added a TODO tag there. I didn’t actually test this because I didn’t think there was a ScalaTest version available.

Reply:
Done